### PR TITLE
Add a check to ensure we have a valid PHP version before loading plugin functionality and output an admin notice f needed

### LINKED ISSUE #129: Add PHP checks
More info: https://github.com/10up/convert-to-blocks/pull/123#issuecomment-1611475826

### PR DESCRIPTION
**Pull request by kmgalanakis**
### Description of the Change

In this PR, I'm adding a check for the minimum required PHP version before attempting to load the plugin files. If the condition is not met an admin notice is displayed and the plugin files are not loaded.

Closes https://github.com/10up/insecure-content-warning/issues/129

### How to test the Change

* To see the admin notice and have the plugin functionality not loaded, try to load the plugin on a site with PHP version lower that 7.4.

### Changelog Entry

> Added - Check for minimum required PHP version before loading the plugin

### Credits

Props @kmgalanakis 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: insecure-content-warning.php [hunk 1]
```diff
@@ -24,6 +24,53 @@
...
+	add_action(
+		'admin_notices',
+		function() {
+			?>
+			<div class="notice notice-error">
+				<p>
+					<?php
+					echo wp_kses_post(
+						sprintf(
+						/* translators: %s: Minimum required PHP version */
```
Comment (peterwilsoncc) [inline insecure-content-warning.php]: Alignment nit: one extra tab to align with line below.

File: insecure-content-warning.php [hunk 2]
```diff
@@ -24,6 +24,53 @@
...
 define( 'INSECURE_CONTENT_VERSION', '1.1.0' );
 
+/**
+ * Get the minimum version of PHP required by this plugin.
+ *
+ * @since 1.1.2
+ *
+ * @return string Minimum version required.
+ */
+function minimum_php_requirement(): string {
```
Comment (peterwilsoncc) [inline insecure-content-warning.php]:
```suggestion
function minimum_php_requirement() {
```

This is to keep the file compatible with PHP 5.6 (as it supports versions of WP prior to 6.3)

File: insecure-content-warning.php [hunk 3]
```diff
@@ -24,6 +24,53 @@
...
+}
+
+/**
+ * Whether PHP installation meets the minimum requirements
+ *
+ * @since 1.1.2
+ *
+ * @return bool True if meets minimum requirements, false otherwise.
+ */
+function site_meets_php_requirements(): bool {
```
Comment (peterwilsoncc) [inline insecure-content-warning.php]:
```suggestion
function site_meets_php_requirements() {
```

**Review by peterwilsoncc — CHANGES_REQUESTED** [review]
A couple of notes about keeping the main file compatible with PHP 5.6 as the plugin supports WP 5.7 and later.

In the linting workflow it would be good to add some compatibility checks:

* for the entire plugin, check that it's compatible with PHP 7.4 and later
* for the main file only, check that it's compatible with PHP 5.6 and later

**Comment by dkotter** [comment]
> A couple of notes about keeping the main file compatible with PHP 5.6 as the plugin supports WP 5.7 and later.

@peterwilsoncc Curious on your reasoning for this? I get that we support WordPress 5.7+ which itself supports PHP 5.6+ but we're also saying we only support PHP 7.4+. While I can see this may be looked at as a conflict (how can we truly claim to support WP 5.7 if we don't support PHP 5.6) I'm personally fine with us pushing users to new PHP versions even if they're running older WP versions.

Is there worry that someone may be running PHP 5.6 and also be running this plugin, even though the plugin only supports PHP 7.4+? I guess to me feels like extra work to try and ensure the main plugin file always stays PHP 5.6 compatible for what is hopefully an edge case (someone running PHP 5.6 that happens to be able to bypass WordPress' update/install requirements and ends up with this plugin installed)

**Comment by peterwilsoncc** [comment]
@dkotter There are two main reasons:

1. If the main file isn't compatible with the older versions of PHP, then the site may fatal before it's shown, in which case including the check seems a little pointless.
2. There's a bug in WP in which the bulk updates screen doesn't check the PHP Compatibility header in the plugin, see [WordPress#59198](https://core.trac.wordpress.org/ticket/59198) so it's possible someone running an older version of WP can upgrade the plugin even though they shouldn't be.

I agree it's an edge case but as it's relatively trivial to protect against I think it's a good idea to do so.

File: insecure-content-warning.php (hunk 2 above)
Comment (kmgalanakis-sage) [reply insecure-content-warning.php]: Done

File: insecure-content-warning.php (hunk 3 above)
Comment (kmgalanakis-sage) [reply insecure-content-warning.php]: Done

**Comment by kmgalanakis-sage** [comment]
Good point @dkotter, thank you for your feedback

File: insecure-content-warning.php (hunk 2 above)
Comment (kmgalanakis) [reply insecure-content-warning.php]: (sorry, wrong profile)

File: insecure-content-warning.php (hunk 3 above)
Comment (kmgalanakis) [reply insecure-content-warning.php]: (sorry, wrong profile)

File: insecure-content-warning.php (hunk 1 above)
Comment (kmgalanakis) [reply insecure-content-warning.php]: Done

**Comment by jeffpaul** [comment]
@peterwilsoncc back to you for review/testing

**Review by peterwilsoncc — APPROVED** [review]
LGM and passed manual testing on different versions of PHP

**Pull request merged**